engine/rig/src/rig.cpp

```diff
@@ -519,14 +533,10 @@ namespace dmRig
             return;
         }
 
-        dmArray<Matrix4>& cache_pose_matrices = context->m_PoseMatrixCache.m_PoseMatrices;
-
-        const uint32_t num_pose_matrices = cache_pose_matrices.Size();
-        EnsureSize(cache_pose_matrices, num_pose_matrices + bone_count);
```

Comment:
Here's the problem. If we add 100 new rig instances in a frame we resize the array 100 times in the first frame. Better to resize the array once outside of the loop over the rig instances (which starts in the Update function)